aws-py-ec2-provisioners/__main__.py

```diff
@@ -12,12 +13,22 @@
 key_name = config.get('keyName')
 public_key = config.get('publicKey')
 
+
 # The privateKey associated with the selected key must be provided (either directly or base64 encoded),
 # along with an optional passphrase if needed.
 def decode_key(key):
+
+    try:
+        key = base64.b64decode(key.encode('ascii')).decode('ascii')
+    except:
+        pass
+
     if key.startswith('-----BEGIN RSA PRIVATE KEY-----'):
         return key
+
     return key.encode('ascii')
```

Comment:
I was confused by along time here, eventually filed https://github.com/pulumi/pulumi/issues/7691 - we do not seem to support `bytes` object, and we were missing b64 decoding step. Integration test passes a value that is a secret string but is actually b64-encoded.